File: core/src/main/java/com/adobe/aem/guides/wknd/spa/angular/core/models/UniversityListComponent.java
package com.adobe.aem.guides.wknd.spa.angular.core.models;

import com.adobe.cq.export.json.ComponentExporter;

import java.io.IOException;
import java.net.URISyntaxException;

public interface UniversityListComponent extends ComponentExporter {
    public University[] getCountry() throws URISyntaxException, IOException;
}

File: ui.frontend/src/app/components/university-list/university-list.component.ts
import { Component, Input, OnInit } from '@angular/core';
import {MapTo} from '@adobe/cq-angular-editable-components';

interface University {
    domains: string[];
    state_province: string;
    country: string;
    alpha_two_code: string;
    web_pages: string[];
    name: string;
}

const CustomEditConfig = {
  emptyLabel: 'University List',
  isEmpty: cqModel =>
    !cqModel || !cqModel.country
};

@Component({
  selector: 'app-custom',
  templateUrl: './university-list.component.html',
  styleUrls: ['./university-list.component.css']
})
export class UniversityListComponent implements OnInit {

  @Input() country: University[];

  constructor() { }

  ngOnInit(): void {
  }

}

MapTo('wknd-spa-angular/components/university-list')(UniversityListComponent, CustomEditConfig);
